R/read_csv.R

```diff
@@ -212,58 +244,94 @@ read_sample_csv <- function(output_files) {
       }
       id <- csv_file_info$id
     }
-    # read sampling data
-    draws <- utils::read.csv(output_file, header = TRUE, comment.char = "#")
+    repaired_model_params <- repair_variable_names(sampling_info$model_params)
+    repaired_sampler_diagnostics <- repair_variable_names(sampling_info$sampler_diagnostics)
+    if (!is.null(parameters)) {
+      for (p in parameters) {
+        if (!(p %in% repaired_model_params)) {
+          stop("The specified parameter ", p, " was not found in the sampling output!")
+        }
+      }
+    }
+    if (!is.null(sampler_diagnostics)) {
+      for (p in sampler_diagnostics) {
+        if (!(p %in% repaired_sampler_diagnostics)) {
+          if (p == "lp__") {
+            stop("lp__ is a part of model parameters not sampler diagnostics.")
+          } else {
+            stop("The specified sampler diagnostic ", p, " was not found in the sampling output!")
+          }
+        }
+      }
+    }
+    if (is.null(col_select)) {
+      col_select <- "lp__"
+      # filter out lp__ to not repeat it
+      parameters <- parameters[parameters!="lp__"]
+      if (is.null(sampler_diagnostics)) {
+        col_select <- c(col_select, sampling_info$sampler_diagnostics)
+      } else {
+        col_select <- c(col_select, unrepair_variable_names(unlist(sampler_diagnostics, use.names=FALSE)))
+      }
+      if (is.null(parameters)) {
+        col_select <- c(col_select, sampling_info$model_params[sampling_info$model_params!="lp__"])
+      } else {
+        col_select <- c(col_select, unrepair_variable_names(unlist(parameters, use.names=FALSE)))
+      }
+      selected_sampler_diagnostics <- sampling_info$sampler_diagnostics[sampling_info$sampler_diagnostics %in% col_select]
+      selected_model_params <- sampling_info$model_params[sampling_info$model_params %in% col_select]
+    }
+    if (length(sampling_info$sampler_diagnostics) == 0 && length(sampling_info$model_params) == 0) {
+      stop("The supplied csv file does not contain any parameter names or data!")
+    }
```

Comment:
I might be missing something, but I think we can throw this error much earlier right after creating `sampling_info`, right?

Reply:
Yeah, moved it to read_sample_info_csv to fail sooner.